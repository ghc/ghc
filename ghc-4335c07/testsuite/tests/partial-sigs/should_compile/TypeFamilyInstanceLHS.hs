{-# LANGUAGE TypeFamilies #-}
module TypeFamilyInstanceLHS where

type family F (a :: *) (b :: *) :: *
type instance F Int  _ = Int
type instance F Bool _ = Bool

foo :: F Int Char -> Int
foo = id
