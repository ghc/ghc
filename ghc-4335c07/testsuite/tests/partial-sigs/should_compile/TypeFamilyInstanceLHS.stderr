TYPE SIGNATURES
  foo :: F Int Char -> Int
TYPE CONSTRUCTORS
  type family F a b :: * open
COERCION AXIOMS
  axiom TypeFamilyInstanceLHS.D:R:FBool_ ::
    F Bool _ = Bool -- Defined at TypeFamilyInstanceLHS.hs:6:15
  axiom TypeFamilyInstanceLHS.D:R:FInt_ ::
    F Int _ = Int -- Defined at TypeFamilyInstanceLHS.hs:5:15
FAMILY INSTANCES
  type instance F Int _
  type instance F Bool _
Dependent modules: []
Dependent packages: [base-4.11.0.0, ghc-prim-0.5.2.0,
                     integer-gmp-1.0.1.0]
